pkg/util/cmapichecker/cmapichecker_test.go

```diff
@@ -0,0 +1,144 @@
+			newError:    errors.New("while creating client: Get \"http://localhost:8080/api?timeout=32s\": dial tcp 127.0.0.1:8080: connect: connection refused"),
+			createError: nil,
+
+			expectedError: ErrAPIServerUnreachable.Error(),
+		},
+		"check API without CRDs installed": {
+			newError:    nil,
+			createError: errors.New("error finding the scope of the object: failed to get restmapping: no matches for kind \"Certificate\" in group \"cert-manager.io\""),
+
+			expectedError: ErrCertManagerCRDsNotFound.Error(),
+		},
+		"check API with webhook service not ready": {
+			newError:    nil,
+			createError: errors.New("Internal error occurred: failed calling webhook \"webhook.cert-manager.io\": Post \"https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s\": service \"cert-manager-webhook\" not found"),
+
+			expectedError: ErrCertManagerAPIEndpointsNotEstablished.Error(),
+		},
+		"check API with webhook pod not accepting connections": {
+			newError:    nil,
+			createError: errors.New("Internal error occurred: failed calling webhook \"webhook.cert-manager.io\": Post \"https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s\": dial tcp 10.96.38.90:443: connect: connection refused"),
+
+			expectedError: ErrWebhookConnectionFailure.Error(),
+		},
+		"check API with webhook certificate not updated in mutation webhook resource definitions": {
+			newError:    nil,
+			createError: errors.New("Internal error occurred: failed calling webhook \"webhook.cert-manager.io\": Post \"https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s\": x509: certificate signed by unknown authority (possibly because of \"x509: ECDSA verification failure\" while trying to verify candidate authority certificate \"cert-manager-webhook-ca\""),
+
+			expectedError: ErrWebhookCertificateFailure.Error(),
+		},
+		"check API with webhook certificate not updated in conversion webhook resource definitions": {
+			newError:    nil,
+			createError: errors.New("conversion webhook for cert-manager.io/v1alpha2, Kind=Certificate failed: Post \"https://cert-manager-webhook.cert-manager.svc:443/convert?timeout=30s\": x509: certificate signed by unknown authority"),
+
+			expectedError: ErrWebhookCertificateFailure.Error(),
+		},
+	}
+
+	for n, test := range tests {
+		t.Run(n, func(t *testing.T) {
+			runTest(t, test)
+		})
+	}
+}
+
+type testT struct {
+	newError    error
+	createError error
+
+	expectedError string
+}
+
+func runTest(t *testing.T, test testT) {
+	errorClient, checker, _ := newFakeCmapiChecker()
+
+	errorClient.newError = test.newError
+	errorClient.createError = test.createError
+
+	err := checker.Check(context.TODO())
+	if err != nil {
+		if err.Error() != test.expectedError {
+			t.Errorf("error differs from expected error:\n%s\n vs \n%s", err.Error(), test.expectedError)
+		}
+	} else {
+		if test.expectedError != "" {
+			t.Errorf("expected error did not occure:\n%s", test.expectedError)
+		}
+	}
+}
```

Comment:
Thanks for writing these tests.
But maybe they could be even more specific if you unit-tested the `translateToSimpleError` function. 
That's only my suggestion.


There are a few lines which are not covered by the tests:

```
go test -coverprofile=cover.out ./pkg/util/cmapichecker/... 
go tool cover -html=cover.out

```


![image](https://user-images.githubusercontent.com/978965/125813327-71e0f346-03ac-4766-8b94-f595f38611f9.png)